src/chocolatey.tests/infrastructure.app/nuget/ChocolateyNugetCredentialProviderSpecs.cs

```diff
@@ -0,0 +1,320 @@
+            protected Uri TargetSourceUri = new Uri(TargetSourceUrl);
+
+            protected NetworkCredential Result;
+
+            private readonly CancellationTokenSource _tokenSource = new CancellationTokenSource();
+            protected CancellationToken CancellationToken
+            {
+                get
+                {
+                    return _tokenSource.Token;
+                }
+            }
+
+            public override void Context()
+            {
+                Configuration = new ChocolateyConfiguration();
+                Provider = new ChocolateyNugetCredentialProvider(Configuration);
+
+                Configuration.Information.IsInteractive = false;
+                Configuration.MachineSources = new List<MachineSourceConfiguration>
+                {
+                    new MachineSourceConfiguration
+                    {
+                        AllowSelfService = true,
+                        VisibleToAdminsOnly = false,
+                        EncryptedPassword = NugetEncryptionUtility.EncryptString("otherPassword"),
+                        Username = "otherUserName",
+                        Key = "https://someotherplace.com/repository/things/",
+                        Name = "not-this-one",
+                        Priority = 1,
+                    },
+                    new MachineSourceConfiguration
+                    {
+                        AllowSelfService = true,
+                        VisibleToAdminsOnly = false,
+                        EncryptedPassword = NugetEncryptionUtility.EncryptString(Password),
+                        Username = Username,
+                        Key = TargetSourceUrl,
+                        Name = TargetSourceName,
+                        Priority = 1,
+                    },
+                };
+            }
+
+            [OneTimeSetUp]
+            public async Task OneTimeSetup()
+            {
+                await With();
+            }
+
+            public virtual async Task With()
+            {
+                var result = await Provider.GetAsync(TargetSourceUri, proxy: null, CredentialRequestType.Unauthorized, message: string.Empty, isRetry: false, nonInteractive: true, CancellationToken);
+                Result = result.Credentials as NetworkCredential;
+            }
+        }
+
+        public class When_using_explicit_credentials_and_source_param : ChocolateyNugetCredentialProviderSpecsBase
+        {
+            public override void Because()
+            {
+                Configuration.Sources = Configuration.ExplicitSources = TargetSourceUrl;
+                Configuration.SourceCommand.Username = "user";
+                Configuration.SourceCommand.Password = "totally_secure_password!!!";
+            }
+
+            [Fact]
+            public void Creates_a_valid_credential()
+            {
+                Result.Should().NotBeNull();
+            }

```

Comment:
This has been updated.

A note on the naming and casing: should it not be PascalCase as well?

Reply:
> A note on the naming and casing: should it not be PascalCase as well?

Not for tests, as the names of tests can become large, it will be easier to read these names when the words are separated by an underscore.